spec/models/bigbluebutton_meeting_spec.rb

```diff
@@ -47,4 +47,176 @@
       end
     end
   end
+
+  describe ".create_meeting_record_from_room" do
+    let(:server) { FactoryGirl.create(:bigbluebutton_server) }
+    let(:room) { FactoryGirl.create(:bigbluebutton_room) }
+
+    context "if there is already a current meeting" do
+      let!(:meeting) { FactoryGirl.create(:bigbluebutton_meeting, room: room, ended: false, running: true, create_time: room.create_time) }
+      subject {
+        expect {
+          BigbluebuttonMeeting.create_meeting_record_from_room(room, {}, server, nil, {})
+        }.not_to change{ BigbluebuttonMeeting.count }
+      }
+      it { BigbluebuttonMeeting.where(room: room).count.should be(1) }
+      it { meeting.reload.running.should be(true) }
+      it { meeting.reload.ended.should be(false) }
+    end
+
+    context "if #create_time is not set in the room" do
+      before { room.update_attributes(create_time: nil) }
+      subject { BigbluebuttonMeeting.create_meeting_record_from_room(room, {}, server, nil, {}) }
+      it("doesn't create a meeting") {
+        BigbluebuttonMeeting.find_by(room_id: room.id).should be_nil
+      }
+    end
+
+    context "if #create_time is set" do
+      let(:user) { FactoryGirl.build(:user) }
+      let(:metadata) {
+        m = {}
+        m[BigbluebuttonRails.configuration.metadata_user_id] = user.id
+        m[BigbluebuttonRails.configuration.metadata_user_name] = user.name
+        m
+      }
+      before {
+        room.create_time = Time.now.utc
+        room.running = !room.running # to change its default value
+        room.record_meeting = !room.record_meeting # to change its default value
+        room.create_time = Time.at(Time.now.to_i - 123)  # to change its default value
+      }
+
+      context "if there's no meeting associated yet creates one" do
+        context "and there's no metadata in the response" do
+          before(:each) {
+            expect {
+              BigbluebuttonMeeting.create_meeting_record_from_room(room, {}, server, nil, {})
```

Comment:
This is called multiple times, one for each `it` block. It would be better to call it just one time within a single `it` block with all tests.